utils/ccloud_library.sh

```diff
@@ -883,7 +880,8 @@ function ccloud::create_ccloud_stack() {
   then
     # Environment is not received so it will be created
     ENVIRONMENT_NAME=${ENVIRONMENT_NAME:-"demo-env-$SERVICE_ACCOUNT_ID"}
-    ENVIRONMENT=$(ccloud::create_and_use_environment $ENVIRONMENT_NAME) 
+    ENVIRONMENT=$(ccloud::create_and_use_environment $ENVIRONMENT_NAME)
+    (($? != 0)) && { echo "$ENVIRONMENT"; exit 1; }
```

Comment:
@javabrett I think this is the reason creation errors weren't surfaced in your environment

Reply:
Should we expand the error message here?

Reply:
@rspurgeon if the status is not 0, the `ENVIRONMENT` _is_ the error message.  Are you saying we should add even more to it?

Reply:
Yes, I'm not sure that's enough to help a user narrow in on the error location for debugging.   Something like "Failed to create (or use) environment: $ENVIRONMENT", or similar

Reply:
If the environment create fails, for example in @javabrett 's test case below, we don't actually show anything here I don't think.  This error message doesn't show the env ID, maybe because it failed.   In his case, the ccloud_stack_create.sh function fails and the ccloud cli provides an error also.  However, we also instruct users to call this function directly by downloading, sourcing, and running this function in bash directly

Reply:
This is the output if env creation fails:

```
[9:12:30] ~/git/examples/ccloud/ccloud-stack(DEVX-2251) ✗: ./ccloud_stack_create.sh
This example uses real Confluent Cloud resources.
To avoid unexpected charges, carefully evaluate the cost of resources before launching the script and ensure all resources are destroyed after you are done running it.
Do you still want to run this script? [y/n] y
Do you also want to create a Confluent Cloud ksqlDB app (hourly charges may apply)? [y/n] n

Creating Confluent Cloud stack for service account demo-app-55, ID: 138905.
Error: unknown shorthand flag: 'o' in -o
Usage:
  ccloud environment [command]

Available Commands:
  create      Create a new Confluent Cloud environment.
  delete      Delete a Confluent Cloud environment and all its resources.
  list        List Confluent Cloud environments.
  update      Update an existing Confluent Cloud environment.
  use         Switch to the specified Confluent Cloud environment.

Global Flags:
  -h, --help            Show help for this command.
  -v, --verbose count   Increase verbosity (-v for warn, -vv for info, -vvv for debug, -vvvv for trace).

Use "ccloud environment [command] --help" for more information about a command.

ERROR: Failed to create environment demo-env-138905. Please troubleshoot (maybe run ./clean.sh) and run again
```

Reply:
I see an issue with the CLI usage, but I was also pointing out that we instruct the users elsewhere to call the function directly.  We should probably clarify the error location, somehow better.

Reply:
In the case where a user has too many environments and calls the function directly:

```
Creating Confluent Cloud stack for service account demo-app-27532, ID: 138915.
Error: CCloud backend error: 1 error occurred:
        * error creating account: Your organization is currently limited to 10 environments


ERROR: Failed to create environment demo-env-138915. Please troubleshoot (maybe run ./clean.sh) and run again
```

So we do show the user the CLI error, but this line doesn't print anything, so maybe we just remove the `echo`?
`(($? != 0)) && { echo "$ENVIRONMENT"; exit 1; }`

Reply:
That echo statement prints:

```
ERROR: Failed to create environment demo-env-138935. Please troubleshoot (maybe run ./clean.sh) and run again
```

Reply:
Ok, thanks I see now.  The overloaded usage of stdout to represent a valid id or an error message as a "return value" confused me.